qa/rpc-tests/test_framework/test_framework.py

```diff
@@ -172,12 +173,15 @@ def main(self):
                 # Dump the end of the debug logs, to aid in debugging rare
                 # travis failures.
                 import glob
-                filenames = glob.glob(self.options.tmpdir + "/node*/regtest/debug.log")
+                filenames = [self.options.tmpdir + "/test_framework.log"]
+                filenames += glob.glob(self.options.tmpdir + "/node*/regtest/debug.log")
                 MAX_LINES_TO_PRINT = 1000
                 for f in filenames:
-                    print("From" , f, ":")
-                    from collections import deque
-                    print("".join(deque(open(f), MAX_LINES_TO_PRINT)))
+                    try:
+                        print("From" , f, ":")
+                        print("".join(deque(open(f), MAX_LINES_TO_PRINT)))
+                    except:
```

Comment:
Printing the tails of the log files here is simply to help if the test case fails on travis. My thinking in adding the try/except handling is that if reading a log file fails for *any* reason, we should just continue and try to print out the other log files. I don't want there to be any chance that this commit makes the logging available on travis any worse.

I've added a print out of the stack trace exception in the except branch. Does that allay your concerns here?